keras_cv/models/segmentation/deeplab.py

```diff
@@ -82,110 +95,88 @@ def __init__(
+
+        output = spatial_pyramid_pooling(feature_map)
+        output = tf.keras.layers.UpSampling2D(
+            size=(height // feature_map.shape[1], width // feature_map.shape[2]),
+            interpolation="bilinear",
+        )(output)
 
-        self._segmentation_head_passed = segmentation_head
         if segmentation_head is None:
-            segmentation_head = tf.keras.Sequential(
-                [
-                    tf.keras.layers.Conv2D(
-                        filters=256,
-                        kernel_size=(1, 1),
-                        padding="same",
-                        use_bias=False,
-                    ),
-                    tf.keras.layers.BatchNormalization(),
-                    tf.keras.layers.Activation("relu"),
-                    tf.keras.layers.Dropout(0.2),
-                    tf.keras.layers.Conv2D(
-                        filters=classes,
-                        kernel_size=(1, 1),
-                        padding="same",
-                        use_bias=False,
-                        activation="softmax",
-                        # Force the dtype of the classification head to float32 to avoid the NAN loss
-                        # issue when used with mixed precision API.
-                        dtype=tf.float32,
-                    ),
-                ]
+            segmentation_head = SegmentationHead(
+                classes=classes,
+                name="segmentation_head",
+                convs=1,
+                dropout=0.2,
+                kernel_size=1,
+                output_activation=segmentation_head_activation,
             )
-        self.segmentation_head = segmentation_head
 
-    def build(self, input_shape):
-        height = input_shape[1]
-        width = input_shape[2]
-        feature_map_shape = self.backbone.compute_output_shape(input_shape)
-        self.up_layer = tf.keras.layers.UpSampling2D(
-            size=(height // feature_map_shape[1], width // feature_map_shape[2]),
-            interpolation="bilinear",
+        # Segmentation head expects a multiple-level output dictionary
+        output = segmentation_head({1: output})
+
+        super().__init__(
+            inputs={
+                "inputs": inputs,
+            },
+            outputs={
+                "output": output,
+            },
+            **kwargs,
         )
 
-    def call(self, inputs, training=None):
-        feature_map = self.backbone(inputs, training=training)
-        output = self.aspp(feature_map, training=training)
-        output = self.up_layer(output, training=training)
-        output = self.segmentation_head(output, training=training)
-        return output
-
-    # TODO(tanzhenyu): consolidate how regularization should be applied to KerasCV.
-    def compile(self, weight_decay=0.0001, **kwargs):
-        self.weight_decay = weight_decay
-        super().compile(**kwargs)
```

Comment:
I think the model will heavily overfit without weight regularization, so we need to keep `compile` and `train_step` here.

Reply:
Adding them back in!
Since these are on new lines, GH might treat them as mine. How can I add you as the author of the methods? 😓

Reply:
I think it's also possible for us to completely rely on `SGD(weight_decay)` -- as long as we can verify this works well on TF 2.10, because the training script requires at least 2.10 as `keras.metrics.MeanIOU` is fixed after 2.10.

Reply:
It would be great if we can do that. 
Should I try it out in the next run or will you? My runs take longer and run for fewer epochs, which makes experiments a bit harder. If we have the time, I don't mind running them.